### PR TITLE
Added new check KIA1601 for ReferenceGrant

### PR DESCRIPTION
**Pull request by hhovsepy**
Issue https://github.com/kiali/kiali/issues/6918

An Error severity message is added for "Spec.From.Namespace" field of "K8sReferenceGrant" object checking if namespace exists: "KIA1601 Namespace is not found or is not accessible"

Screenshot from 2024-01-08 11-12-03

### CODE REVIEW

File: business/checkers/k8sreferencegrants/namespace_checker.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"fmt"
+
+	k8s_networking_v1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/kiali/kiali/models"
+)
+
+type NamespaceChecker struct {
+	ReferenceGrant k8s_networking_v1beta1.ReferenceGrant
```
Comment (josunect) [inline business/checkers/k8sreferencegrants/namespace_checker.go:12]: That could be alphabetized?

**Review by josunect — COMMENTED** [review]
image
 It works as expected and code looks good, just a small comment. 

Also, when saving the yaml creating an incorrect namespace, this is the message received: 

image

I think it could be a bit more clear with something like: 

"foo Namespace is not found or is not accessible"

**Comment by hhovsepy** [comment]
I think this is working by design of validation notifications, the messages does not contain `{}` variables to replace with strings. They are shown in a format `type:name message`, I agree this message format is a bit outdated. But this can be changed for all validations in a scope of a separate epic.
> 
> Also, when saving the yaml creating an incorrect namespace, this is the message received:
> 
> image
> 
> I think it could be a bit more clear with something like:
> 
> "foo Namespace is not found or is not accessible"

**Comment by josunect** [comment]
> I think this is working by design of validation notifications, the messages does not contain `{}` variables to replace with strings. They are shown in a format `type:name message`, I agree this message format is a bit outdated. But this can be changed for all validations in a scope of a separate epic.
> 
> > Also, when saving the yaml creating an incorrect namespace, this is the message received:
> > image
> > I think it could be a bit more clear with something like:
> > "foo Namespace is not found or is not accessible"

Thanks for clarifying, @hhovsepy !

**Pull request merged**